cgi/user2.pl

```diff
@@ -0,0 +1,361 @@
+					type => "password",
+					label2 => "password"
+				},
+				{
+					field => "confirm_password",
+					type => "password",
+					label2 => "password_confirm"
+				},
+			]
+		};
+
+		# Professional account
+		push @{$template_data_ref->{sections}}, {
+			id => "professional",
+			fields => [
+				{
+					field => "pro",
+					type => "checkbox",
+				},
+				{
+					field => "pro_checkbox",
+					type => "hidden",
+					value => 1,
+				},
+				{
+					field => "requested_org",
+				}
+			]
+		};
+
+		my $team_section_ref = { id => "teams", fields => [] };
+		for (my $i = 1; $i <= 3; $i++) {
+			push @{$team_section_ref->{fields}}, {
+				field => "team_". $i,
+				label2 => sprintf(lang("team_s"), $i),
+			 };
+		};
+
+		push @{$template_data_ref->{sections}}, {%$team_section_ref};
+
+		my $administrator_section_ref = { id => "administrator", fields => [] };
+		push  @{$administrator_section_ref->{fields}}, {
+			field => "org",
+		};
+		foreach my $group (@user_groups) {
+			push @{$administrator_section_ref->{fields}}, {
+				field =>  "user_group_". $group,
+				label2 =>  "user_group_" . ${group} . "_description",
+				type => "checkbox",
+				value => $user_ref->{$group},
+			};
+		};
+
+		push @{$template_data_ref->{sections}}, {%$administrator_section_ref};
+	}
+
+	# Add labels, types, descriptions, notes and existing values for all fields
+	foreach my $section_ref (@{$template_data_ref->{sections}}) {
+
+		foreach my $field_ref (@{$section_ref->{fields}}) {
+
+			my $field = $field_ref->{field};
+
+			# Default to text field
+			if (not defined $field_ref->{type}) {
+				$field_ref->{type} = "text";
+			}
+
+			# id to use for lang() strings
+			my $field_lang_id = $field;
+
+			if (not defined $field_ref->{value}) {
+				$field_ref->{value} = $user_ref->{$field};
+			};
+
+			# Label
+			if (not defined $field_ref->{label}) {
+				$field_ref->{label} = lang($field_lang_id);
+			}
+			else {
+				$field_ref->{label} = $field_lang_id;
+			}
+		};
+	};
+
+	$template_data_ref->{accepted_organization} = $user_ref->{org};
+	$template_data_ref->{pro_account_org} = sprintf(lang("this_is_a_pro_account_for_org"),"<b>" . $user_ref->{org} . "</b>");
+
+	my $pro_checked = '';
+
+	# Check the "pro account" checkbox for register screen on the producers platform
+
+	if (((defined $user_ref->{pro}) and ($user_ref->{pro}))
+		or ((defined $server_options{producers_platform}) and ($type eq "add"))) {
+		$pro_checked = "on";
+	}
+
+	$template_data_ref->{product_type} = $options{product_type};
+	$template_data_ref->{pro_checked} = $pro_checked;
+	my $requested_org_ref = retrieve_org($user_ref->{requested_org});
+	$template_data_ref->{requested_org_ref} = $requested_org_ref;
+	$template_data_ref-> {org_name} = sprintf(lang("add_user_existing_org"), org_name($requested_org_ref));
+	$template_data_ref->{teams_flag} = not ((defined $server_options{private_products}) and ($server_options{private_products}));
+	$template_data_ref->{admin_flag} = $admin;
```

Comment:
There is already a "admin" variable that is automatically set by process_template():

sub process_template($$$) {
	
	my $template_filename = shift;
	my $template_data_ref = shift;
	my $result_content_ref = shift;
	
	# Add functions and values that are passed to all templates

	$template_data_ref->{product_type} = $options{product_type};
	$template_data_ref->{admin} = $admin;

Reply:
Alright, `$admin` was causing me a lot of issues, and I did not where it was being set. That is why I changed it to `$admin_flag`. Let me check it again how can I fix it.